src/bcWithoutPred+Pred.stan

```diff
@@ -0,0 +1,226 @@
+      K_div_y = mdivide_right_tri_low(L_K_div_y', L_K)';
+
+	  // Evaluate s21 s12
+      for (i in 1:N) {
+        for (j in 1:n_pred) {
+          k_x1_x2[i, j] = exp(-dot_self((xt[i] - xt_pred[j]) .* beta_eta)) / lambda_eta;
+        }
+      }
+      for (i in 1:n) {
+        for (j in 1:n_pred) {
+          k_x1_x2[i, j] = k_x1_x2[i, j] + exp(-dot_self((xt[i][1:p] - xt_pred[j][1:p]) .* beta_delta)) / lambda_delta;
+        }
+      }
+      y_pred_mu = (k_x1_x2' * K_div_y);
+	  
+	  // v_pred' * v_pred = s21 s11{-1} s12
+      v_pred = mdivide_left_tri_low(L_K, k_x1_x2);
+
+	  // Evaluate s22
+      for (i in 1:(n_pred-1)) {
+        sigma_eta_pred[i, i] = 1/lambda_eta + sigma;
+        for (j in (i+1):n_pred) {
+          sigma_eta_pred[i, j] = exp(-dot_self((xt_pred[i] - xt_pred[j]) .* beta_eta))/lambda_eta;
+          sigma_eta_pred[j, i] = sigma_eta_pred[i, j];
+        }
+      }
+      sigma_eta_pred[n_pred, n_pred] = 1/lambda_eta + sigma;
+
+      for (i in 1:(n_pred-1)) {
+        sigma_delta_pred[i, i] = 1/lambda_delta + 1e-9;
+        for (j in (i+1):n_pred) {
+          sigma_delta_pred[i, j] = exp(-dot_self((xt_pred[i][1:p] - xt_pred[j][1:p]) .* beta_delta))/lambda_delta;
+          sigma_delta_pred[j, i] = sigma_delta_pred[i, j];
+        }
+      }
+      sigma_delta_pred[n_pred, n_pred] = 1/lambda_delta + 1e-9;
+
+      cov_y_pred = sigma_eta_pred + sigma_delta_pred;
+	  
+	  // Evaluate COV{y(x_pred, tf)|y,.} = s22 - s21 s11{-1} s12
+      cov_y_pred = cov_y_pred - v_pred' * v_pred + diag_matrix(rep_vector(1e-9, n_pred));
+      y_pred = multi_normal_rng(y_pred_mu, cov_y_pred);
+    }
+    return y_pred;
+  }
+}
+data {
+  int<lower=0> n; // number of field data
+  int<lower=0> m; // number of computer simulation
+  int<lower=1> n_pred; // number of predictions
+  int<lower=0> p; // number of observable inputs x
+  int<lower=0> q; // number of calibration parameters t
+  vector[n] y; // field observations
+  vector[m] eta; // output of computer simulations
+  row_vector[p] xf[n]; // observable inputs corresponding to y
+  // (xc, tc): design points corresponding to eta
+  row_vector[p] xc[m]; 
+  row_vector[q] tc[m]; 
+  row_vector[p] x_pred[n_pred]; 
+}
+
+transformed data {
+  int<lower=1> N;
+  vector[n+m] z; // z = [y, eta]
+  vector[n+m] mu; // mean vector
+
+  N = n + m;
+  // set mean vector to zero
+  for (i in 1:N) {
+    mu[i] = 0;
+  }
+  z = append_row(y, eta); 
+}
+
+parameters {
+  // tf: calibration parameters
+  // rho_eta: reparameterization of beta_eta
+  // rho_delta: reparameterization of beta_delta
+  // lambda_eta: precision parameter for eta
+  // lambda_delta: precision parameter for bias term
+  // lambda_e: precision parameter of observation error
+  row_vector<lower=0,upper=1>[q] tf; 
+  row_vector<lower=0,upper=1>[p+q] rho_eta;
+  row_vector<lower=0,upper=1>[p] rho_delta;
+  real<lower=0> lambda_eta; 
+  real<lower=0> lambda_delta; 
+  real<lower=0> sigma;
```

Comment:
I'm confused by your use of sigma and delta.
Wouldn't sigma = 1/lambda_eta and delta is just a constant used to make the covariance matrix positive definite.